openapi/openapi/src/main/scala/endpoints4s/openapi/JsonSchemas.scala

```diff
@@ -325,15 +457,15 @@ trait JsonSchemas extends algebra.JsonSchemas with TuplesSchemas {
       schema: JsonSchema[A],
       example: A
   ): JsonSchema[A] = {
-    val exampleJson = schema.ujsonSchema.codec.encode(example)
+    lazy val exampleJson = schema.ujsonSchema.codec.encode(example)
     def updatedDocs(djs: DocumentedJsonSchema): DocumentedJsonSchema =
       djs match {
-        case s: DocumentedRecord => s.copy(example = Some(exampleJson))
-        case s: DocumentedCoProd => s.copy(example = Some(exampleJson))
+        case s: DocumentedRecord => s.withExample(exampleJson)
+        case s: DocumentedCoProd => s.withExample(exampleJson)
         case s: Primitive        => s.copy(example = Some(exampleJson))
         case s: Array            => s.copy(example = Some(exampleJson))
         case s: DocumentedEnum   => s.copy(example = Some(exampleJson))
-        case s: LazySchema       => LazySchema(updatedDocs(s.value))
+        case s: LazySchema       => LazySchema(s.name, updatedDocs(s.value))
```

Comment:
Why should `LazySchema` be handled as a case in this match, but `DocumentedRecord.Lazy` and `DocumentedCoProd.Lazy` are not? What is the fundamental conceptual difference? It would be really nice if there were some unified handling of lazy schemas.

Reply:
`DocumentedRecord.Lazy` and `DocumentedCoProd.Lazy` are handled by the case on `DocumentedRecord` and `DocumentedCoProd`. They cannot by handled the same way as `LazySchema` because `updatedDocs(s.value)` returns a `DocumentedJsonSchema` while `DocumentedRecord.Lazy` requires a `DocumentedRecord` so a case like the following doesn't work:

```
  case s: DocumentedRecord.Lazy => DocumentedRecord.Lazy(s.name, updatedDocs(s.value))
```

This is why I added the `withName`, `withTitle` etc. on `DocumentedRecord` and `DocumentedTagged`. I guess I could also add these methods on `LazySchema`, or even on `DocumentedJsonSchema` so we wouldn't even need these pattern matchings.

WDYT?